File: api/src/main/java/com/github/av2/api/model/Delivery.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Delivery information")
public class Delivery {
    @Schema(description = "The unique identifier for the delivery", required = true)
    private Integer deliveryId;
    
    @Schema(description = "The ID of the supplier providing this delivery", required = true)
    private Integer supplierId;
    
    @Schema(description = "Delivery date and time")
    private String deliveryDate;
    
    @Schema(description = "The name of the delivery", required = true)
    private String name;
    
    @Schema(description = "Description of the delivery")
    private String description;
    
    @Schema(description = "The status of the delivery", required = true)
    private String status;
}

File: api/src/models/delivery.ts
/**
 * @swagger
 * components:
 *   schemas:
 *     Delivery:
 *       type: object
 *       required:
 *         - deliveryId
 *         - orderId
 *         - status
 *       properties:
 *         deliveryId:
 *           type: integer
 *           description: The unique identifier for the delivery
 *         orderId:
 *           type: integer
 *           description: The ID of the order being delivered
 *         status:
 *           type: string
 *           description: Current status of the delivery
 *           enum: [pending, in-transit, delivered, failed]
 *         scheduledDate:
 *           type: string
 *           format: date-time
 *           description: Scheduled delivery date and time
 *         actualDeliveryDate:
 *           type: string
 *           format: date-time
 *           description: Actual delivery date and time
 *         notes:
 *           type: string
 *           description: Additional notes about the delivery
 */
export interface Delivery {
  deliveryId: number;
  supplierId: number;
  deliveryDate: string;
  name: string;
  description: string;
  status: string;
}
